Sources/NostrSDK/EventCreating.swift

```diff
@@ -291,6 +291,67 @@ public extension EventCreating {
         return try MuteListEvent(content: encryptedContent ?? "", tags: publicTags, signedBy: keypair)
     }
     
+    /// Creates a ``BookmarksListEvent`` (kind 10003) containing an uncategorized, "global" list of things a user wants to save.
+    /// - Parameters:
+    ///   - publiclyBookmarkedEventIds: Event ids to bookmark.
+    ///   - privatelyBookmarkedEventIds: Event ids to privately bookmark.
+    ///   - publiclyBookmarkedArticlesCoordinates: Articles coordinates to bookmark.
+    ///   - privatelyBookmarkedArticlesCoordinates: Articles coordinates to privately bookmark.
+    ///   - publiclyBookmarkedHashtags: Hashtags to bookmark.
+    ///   - privatelyBookmarkedHashtags: Hashtags to privately bookmark.
+    ///   - publiclyBookmarkedLinks: Links to bookmark.
+    ///   - privatelyBookmarkedLinks: Links to privately bookmark.
+    func bookmarksList(withPubliclyBookmarksEventIds publiclyBookmarkedEventIds: [String] = [],
+                       privatelyBookmarkedEventIds: [String] = [],
+                       publiclyBookmarkedArticlesCoordinates: [EventCoordinates] = [],
+                       privatelyBookmarkedArticlesCoordinates: [EventCoordinates] = [],
+                       publiclyBookmarkedHashtags: [String] = [],
+                       privatelyBookmarkedHashtags: [String] = [],
+                       publiclyBookmarkedLinks: [URL] = [],
+                       privatelyBookmarkedLinks: [URL] = [],
+                       signedBy keypair: Keypair) throws -> BookmarksListEvent {
+        var publicTags = [Tag]()
+        
+        for eventId in publiclyBookmarkedEventIds {
+            publicTags.append(.event(eventId))
+        }
+        for coordinates in publiclyBookmarkedArticlesCoordinates {
+            publicTags.append(coordinates.tag)
+        }
+        for hashtag in publiclyBookmarkedHashtags {
+            publicTags.append(.hashtag(hashtag))
+        }
+        for link in publiclyBookmarkedLinks {
+            publicTags.append(Tag(name: .webURL, value: link.absoluteString))
+        }
+        
+        var secretTags = [[String]]()
+        for eventId in privatelyBookmarkedEventIds {
+            secretTags.append([TagName.event.rawValue, eventId])
```

Comment:
I'm not sure what you mean. For these standard lists, the secret tags get encrypted all together at the end.

Reply:
Oh I see what you mean. Yeah the NIP doesn't have any examples of adding the relay hint to the "e" tags, but I guess it could be there. I'm not sure how you would let the call site specify pairs like that unless you just made them build all the tags manually.

Reply:
Yeah, they'd have to build them manually. But I think we can maybe hide some of that behind an abstraction by building a second layer on top that is relay aware.